crates/scheduler/src/producer.rs

```diff
@@ -25,9 +25,15 @@ pub async fn start_producer<T>(
 where
     T: SchedulerAppState,
 {
-    use rand::Rng;
-    let timeout = rand::thread_rng().gen_range(0..=scheduler_settings.loop_interval);
-    tokio::time::sleep(std::time::Duration::from_millis(timeout)).await;
+    use std::time::Duration;
+
+    use rand::distributions::{Distribution, Uniform};
+
+    let mut rng = rand::thread_rng();
+    let timeout = Uniform::try_from(0..=scheduler_settings.loop_interval)
+        .into_report()
+        .change_context(errors::ProcessTrackerError::InvalidInput)?;
```

Comment:
Could you please raise `ConfigurationError` instead, as suggested in https://github.com/juspay/hyperswitch/pull/1908#discussion_r1296186188? (And remove the `InvalidInput` enum variant?)